### PR TITLE
adds beginnings of fdk-go

### PR DESCRIPTION
**Pull request by rdallman**
this has as much as we support for now and should be the base minimum to use an fdk. i wrote some simple tests to make sure that things basically work. the README should give an idea of how things look. 

we can add other stuff as we go, open to design feedback, but really want to make it use stdlib parts in the function signature for Handler so that users can test their handlers on their own really easily without any fn parts.

### CODE REVIEW

**Comment by rdallman** [comment]
@treeder @ others ?

File: README.md [hunk 1]
```diff
@@ -45,36 +45,53 @@ import (
 )
 
 func main() {
-  fdk.Do(myHandler)
+  fdk.Do(new(mydoer))
 }
 
-func myHandler(ctx context.Context, in io.Reader, out io.Writer) error {
+// use a struct to maintain state between invocations
```
Comment (treeder) [inline README.md]: Should we promote using global state?  I get that they *can* do this and have some temporary global state, but should we promote it?

File: README.md [hunk 2]
```diff
@@ -18,10 +18,10 @@ import (
 )
 
 func main() {
-  fdk.Do(myHandler)
+  fdk.Do(fdk.HandlerFunc(myHandler))
```
Comment (treeder) [inline README.md]: Could we call it `fdk.Handle()` instead of `Do()`

File: README.md (hunk 1 above)
Comment (rdallman) [reply README.md]: by having our integration as an interface (vs a `func` type) we are expressly promoting them not to have global state, I would think, at least in the sense of package level variables? I mostly followed the go `net/http` conventions, which seem familiar for most go developers and allow a lot of flexibility for not having state / having state if they'd like.

File: README.md (hunk 2 above)
Comment (rdallman) [reply README.md]: sure. `Serve` seems apt, as well. will go with Handle for now barring other feedback

**Comment by rdallman** [comment]
> And we should probably make all basic examples use JSON input and output.

roger

**Comment by rdallman** [comment]
i'd like it to implement `http.Handler` but it seems we want to abstract that away (even though, really, this is what the users are doing precisely with `fn` as an intermediary). would be kind of cool to think that devs could just use their existing handlers in some of their http services without having to make any changes though.

File: README.md (hunk 1 above)
Comment (rdallman) [reply README.md]: i guess maybe putting an explicit example using a struct in the README is unnecessary, was just trying to show it I guess (it's contrived, I agree) -- I'll remove from README

**Comment by rdallman** [comment]
ok, updated https://github.com/fnproject/fdk-go/pull/1/commits/270eb628aaaa5050ea728a9b978bd8cb7ecb1235 @treeder 

will work on a bloggo

**Pull request merged**